src/Compilers/Core/MSBuildTask/GenerateMSBuildEditorConfig.cs

```diff
@@ -98,7 +107,33 @@ public override bool Execute()
             }
 
             ConfigFileContents = builder.ToString();
-            return true;
+            return WriteMSBuildEditorConfig();
+        }
+
+        public bool WriteMSBuildEditorConfig()
+        {
+            try
+            {
+                if (File.Exists(FileName.ItemSpec))
+                {
+                    string existingContents = File.ReadAllText(FileName.ItemSpec);
+                    if (existingContents.Length == ConfigFileContents.Length)
```

Comment:
>if (existingContents.Length == ConfigFileContents.Length) [](http://example.com/codeflow?start=20&length=57)

Is this check necessary, given the next check compares the values?

Reply:
Think we can remove this because the implementation of `string.Equals` begins by a length check on the two values

https://source.dot.net/#System.Private.CoreLib/String.Comparison.cs,31b307b02a3bd6b9